webui/react/src/components/CheckpointModal.tsx

```diff
@@ -121,21 +127,28 @@ const CheckpointModal: React.FC<Props> = (
       width={768}
       onCancel={onHide}>
       <div className={css.base}>
-        {renderRow(
-          'Source', (
-            <div className={css.source}>
-              <Link path={paths.experimentDetails(checkpoint.experimentId)}>
-                Experiment {checkpoint.experimentId}
-              </Link>
-              <span className={css.sourceDivider} />
-              <Link path={paths.trialDetails(checkpoint.trialId, checkpoint.experimentId)}>
-                Trial {checkpoint.trialId}
-              </Link>
-              <span className={css.sourceDivider} />
-              <span>Batch {totalBatchesProcessed}</span>
-            </div>
-          ),
-        )}
+        {renderRow('Source', checkpoint.experimentId ? (
+          <div className={css.source}>
+            <Link path={paths.experimentDetails(checkpoint.experimentId.toString())}>
+              Experiment {checkpoint.experimentId}
+            </Link>
+            {!!checkpoint.trialId && (
+              <><span className={css.sourceDivider} />
+                <Link path={paths.trialDetails(checkpoint.trialId, checkpoint.experimentId)}>
+                  Trial {checkpoint.trialId}
+                </Link>
+              </>
+            )}
+            {!!checkpoint.totalBatches && (
```

Comment:
just checking that we don't expect a totalBatches of `0` to render this row. it probably wouldn't make sense for it anyway

Reply:
No, I am treating `0` as a null value for `totalBatches`. Is there a preferred way to indicate this? Explicit `>`?